osv/ecosystems/bioconductor.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Bioconductor helpers."""
+
+import requests
+
+from . import config
+from .helper_base import Ecosystem, EnumerateError
+from .. import semver_index
+
+class Bioconductor(Ecosystem):
+  """Bioconductor ecosystem helpers."""
+
+  # Use the Posit Public Package Manager API to pull both the current and older versions for a specific package
+  # since Bioconductor doesn't natively support this functionality.
+  _API_PACKAGE_URL_POSIT_BIOCONDUCTOR = 'https://packagemanager.posit.co/__api__/repos/4/packages/{package}?bioc_version={bioc_version}'
+  _API_BIOC_VERSIONS_URL = 'https://packagemanager.posit.co/__api__/status'
+  _BIOC_VERSIONS = ['3.18', '3.17', '3.16', '3.15', '3.14', '3.13', '3.12', '3.11', '3.10', '3.9', '3.8', '3.7', '3.6', '3.5', '3.4', '3.3', '3.2', '3.1']
+
+  def __init__(self, *args, **kwargs):
+    # Call the init of Ecosystem class
+    super().__init__(*args, **kwargs)
+
+    # Refresh the versions
+    self.refresh_bioc_versions()
+
+  def refresh_bioc_versions(self):
+    """
+    Pull latest Bioconductor versions
+    """
+    response = requests.get(self._API_BIOC_VERSIONS_URL)
+    data = response.json()
+
+    self._BIOC_VERSIONS = [bioc['bioc_version'] for bioc in data['bioc_versions']]
+
+  def sort_key(self, version):
```

Comment:
Could you not then subclass `SemverEcosystem` instead of `Ecosystem` making this not necessary?

Reply:
I guess if Bioconductor is guaranteed to be semver-compliant you can forgo this whole file and just use the `SemverEcoystem` as others do: https://github.com/google/osv.dev/blob/434a7f55b59f186f3f32599a809842c6ce9d4369/osv/ecosystems/_ecosystems.py#L36-L39
https://github.com/google/osv.dev/blob/434a7f55b59f186f3f32599a809842c6ce9d4369/osv/ecosystems/_ecosystems.py#L58-L63